### PR TITLE
Various changes to tsctl and the API

### PR DESCRIPTION
**Pull request by kiddinn**
This changes few things:

1. Adds version information to tsctl, #1268
2. Adds version information to the API and adds a query to the API client to look up the backend version.
3. Adding status check to `tsctl list_sketches`, #1269
4. Adding a `description` property to the index API client object.
5. Adding some error handling into the API for fetching sketch to deal with potentially missing searchindexes in timelines
6. Adding some error handling into the `add_index` option in tsctl, see #1270
7. Changes to how searchindexes are deleted in the API 
8. Some other error handling or checks added to the API.

### CODE REVIEW

File: timesketch/api/v1/resources/searchindex.py [hunk 1]
```diff
@@ -126,6 +126,7 @@ def delete(self, searchindex_id):
                 HTTP_STATUS_CODE_FORBIDDEN,
                 '\n'.join(error_strings))
 
+        searchindex.set_status(status='deleted')
         db_session.delete(searchindex)
```
Comment (berggren) [inline timesketch/api/v1/resources/searchindex.py:130]: Why setting the status when you delete the object next?

Comment (berggren) [reply timesketch/api/v1/resources/searchindex.py:130]: Do we really want to delete the object from the datastore? Or just mark it as deleted? This is what happens when you delete a sketch. It is not fully deleted from SQL, just marked as such.

File: timesketch/api/v1/routes.py [hunk 2]
```diff
@@ -33,6 +33,7 @@
 from .resources.sketch import SketchResource
 from .resources.sketch import SketchListResource
 from .resources.archive import SketchArchiveResource
+from .resources.information import TSVersionResource
```
Comment (berggren) [inline timesketch/api/v1/routes.py:36]: IMO, rename to VersionResource

File: timesketch/api/v1/resources/searchindex.py (hunk 1 above)
Comment (kiddinn) [reply timesketch/api/v1/resources/searchindex.py:130]: this is the behavior as it has been so far... but that is a good question, I think we should just rather close the index and mark it as deleted.

File: timesketch/api/v1/routes.py (hunk 2 above)
Comment (kiddinn) [reply timesketch/api/v1/routes.py:36]: agree

**Pull request merged**